### PR TITLE
themes/pzq: Add pzq themes inspired by ys-theme from oh-my-zsh and rana

### PR DESCRIPTION
**Pull request by ZQPei**
* Add pzq theme, which is inspired on ys-theme from oh-my-zsh and rana theme
* The pzq theme solves the problem of misplaced commands in rena

### CODE REVIEW

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:47]:
```suggestion
  local code=$?
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:93]: Do we want to expose the value to users? If not, we can make it local.

```suggestion
  local MAX_PWD_LENGTH=20
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:96]:
```suggestion
  local RELATIVE_PWD=${PWD/#$HOME/\~}
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:103]:
```suggestion
      local TRUNCATED_PWD=${RELATIVE_PWD:$offset:$MAX_PWD_LENGTH}
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:113]:
```suggestion
    local virtualenv=$(basename "$VIRTUAL_ENV")
```

Prefer `$(...)` to ``` `...` ``` which is deprecated by POSIX.

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:115]:
```suggestion
    local VIRTUALENV_THEME_PROMPT_PREFIX='('
    local VIRTUALENV_THEME_PROMPT_SUFFIX=')'
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:138]:
```suggestion
  if [ "$OSTYPE" = "linux-gnu" ];
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:122]:
```suggestion
    local CONDAENV_THEME_PROMPT_PREFIX='('
    local CONDAENV_THEME_PROMPT_SUFFIX=')'
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:131]: Modern terminals may not support these SCO sequences. Most terminals support `\e7` and `\e8`, respectively. Also, by using `\[...\]`, we need to let Bash know that the cursor movements between these two values are canceled.

```suggestion
  local SAVE_CURSOR='\[\e7'
  local RESTORE_CURSOR='\e8\]'
```

Or, if you want to make them work in a wider range of terminals, you could obtain them through `tput sc` and `tput rc`, respectively.

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:132]: 500 might not be sufficient for this 4K era...

```suggestion
  local MOVE_CURSOR_RIGHTMOST='\033['${COLUMNS:-9999}'C'
```

File: themes/pzq/pzq.theme.sh
Comment (akinomyoga) [inline themes/pzq/pzq.theme.sh:31]:
```suggestion
      TITLEBAR="\[\033]0;\w\e\\\]"
```

- OSC terminated by BEL is deprecated. It is recommended to use the ANSI form terminated by ST (`\e\\`).
- We need to enclose it with `\[...\]` to let Bash know that the cursor will not be moved by the sequence.

**Review by akinomyoga — CHANGES_REQUESTED** [review]

**Comment by akinomyoga** [comment]
Please also add an image of the theme and update [themes/THEMES.md](https://github.com/ohmybash/oh-my-bash/blob/master/themes/THEMES.md)!

**Comment by akinomyoga** [comment]
I'm thinking of taking over this PR. If you still would like to work on it by yourself, please let me know.

**Review by akinomyoga — APPROVED** [review]

**Comment by akinomyoga** [comment]
I have rebased the PR on top of the latest master, applied suggested changes, and added other fixes. I have also added an image of the theme in `themes/THEMES.md`. I'll keep this PR for about one week and am going to merge it if there is no further comments. @nntoan If you would like to review the PR, please let me know, in which case I'll wait for your review before merging.

**Comment by ZQPei** [comment]
LGTM

**Comment by akinomyoga** [comment]
@ZQPei Thank you for checking the changes to the PR!

**Pull request merged**
